apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.InsertData.cs

```diff
@@ -136,39 +135,20 @@ private BigQueryJob UploadData(JobConfigurationLoad loadConfiguration, Stream in
         }
 
         /// <inheritdoc />
-        public override void InsertRows(TableReference tableReference, IEnumerable<BigQueryInsertRow> rows, InsertOptions options = null)
+        public override BigQueryInsertResults InsertRows(TableReference tableReference, IEnumerable<BigQueryInsertRow> rows, InsertOptions options = null)
         {
             var request = CreateInsertAllRequest(tableReference, rows, options, out bool hasRows);
             if (!hasRows)
             {
-                return;
+                return new BigQueryInsertResults(this, options, rows, new TableDataInsertAllResponse());
             }
-            var response = request.Execute();
-            HandleInsertAllResponse(response, options);
-        }
-
-        private void HandleInsertAllResponse(TableDataInsertAllResponse response, InsertOptions options)
-        {
-            var errors = response.InsertErrors;
-            bool shouldThrow = options == null || !options.SuppressInsertErrors;
-            if (errors?.Count > 0 && shouldThrow)
+            TableDataInsertAllResponse response = request.Execute();
+            BigQueryInsertResults result = new BigQueryInsertResults(this, options, rows, response);
+            if(options?.SuppressInsertErrors == true)
```

Comment:
`return options?.SuppressInsertErrors == true ? result : result.ThrowOnAnyError();`

Reply:
Or move it into an internal helper method in BigQueryInsertResults accepting a `bool?`, and call it from here and the async one.